services/processor/tasks/power/power.go

```diff
@@ -112,51 +112,86 @@ func (ppt *ProcessPowerTask) ParseArgs(job *work.Job) error {
 		return err
 	}
 
-	ppt.maddr = maddr
-	ppt.head = mhead
-	ppt.tsKey = tsKey
-	ppt.ptsKey = ptsKey
-	ppt.stateroot = mstateroot
+	p.maddr = maddr
+	p.head = mhead
+	p.tsKey = tsKey
+	p.ptsKey = ptsKey
+	p.stateroot = mstateroot
 	return nil
 }
 
-func (ppt *ProcessPowerTask) Task(job *work.Job) error {
-	if err := ppt.ParseArgs(job); err != nil {
+func (p *ProcessPowerTask) Task(job *work.Job) error {
+	if err := p.ParseArgs(job); err != nil {
 		return err
 	}
 	ctx := context.Background()
 	ctx, span := global.Tracer("").Start(ctx, "ProcessPowerTask.Task")
 	defer span.End()
 
-	powerActor, err := ppt.node.StateGetActor(ctx, builtin.StoragePowerActorAddr, ppt.tsKey)
+	powerActor, err := p.node.StateGetActor(ctx, builtin.StoragePowerActorAddr, p.tsKey)
 	if err != nil {
-		return err
+		return xerrors.Errorf("loading power actor: %w", err)
 	}
 
-	powerStateRaw, err := ppt.node.ChainReadObj(ctx, powerActor.Head)
+	pstate, err := power.Load(p.node.Store(), powerActor)
 	if err != nil {
-		return err
+		return xerrors.Errorf("loading power actor state: %w", err)
 	}
 
-	var powerActorState power.State
-	if err := powerActorState.UnmarshalCBOR(bytes.NewReader(powerStateRaw)); err != nil {
+	/*
+		TODO blocked on: https://github.com/filecoin-project/lotus/pull/4079/files
```

Comment:
Waiting on https://github.com/filecoin-project/lotus/pull/4079, we can merge without it and this _should_ all still work